server/container_create_linux.go

```diff
@@ -290,6 +291,11 @@ func (s *Server) createSandboxContainer(ctx context.Context, ctr ctrfactory.Cont
 		}
 	}()
 
+	// Avoid image removal during container creation
+	imageRemovalLock := s.StorageImageServer().ImageRemovalLock(imageID)
+	imageRemovalLock.RLock()
+	defer imageRemovalLock.RUnlock()
+
```

Comment:
i don't know if we need this here. The StorageRuntimeServer().CreateContainer() will mount the image into the container's new rootfs, so a RemoveImage request will fail (AFAIU). Why do we need it after that?

Reply:
I think I misunderstood https://github.com/cri-o/cri-o/pull/7850#discussion_r1514831228, moved it back to the previous location.

I did a test with a `time.Sleep(…)` and was able to produce an image removal during container creation. With the code in this PR, that issue was fixed by waiting on the deletion and then returning the right error (image in use by container).